### ISSUE TITLE
is_resource - RedundantCondition error.

### ISSUE THREAD

**Issue by Warxcell** [vimeo/psalm#5854]
Following [snippet](https://psalm.dev/r/c9a6fb64de) throws error 
`ERROR: RedundantCondition - 13:5 - Type resource for $stream is always resource`

But in fact `$stream` is closed in other function. And that function could not be even available for PSALM ro analyse it.

According PHP docs:

```
Note:

is_resource() is not a strict type-checking method: it will return false if value is a resource variable that has been closed.
```
In my opinion: this shouldn't throw error at all, unless it could analyse code and to guarantee that stream is not closed anywhere.

**Comment by muglug** [comment]
Given that resources are effectively on their way out I won't be fixing this, but you can work around it like so: https://psalm.dev/r/4992a1fdf4
